### PR TITLE
Managing a group's members list

### PR DESCRIPTION
**Pull request by juhangsin**

### CODE REVIEW

File: src/group/controller/collection.ts [hunk 1]
```diff
@@ -23,4 +23,4 @@ class GroupMemberCollectionControlller extends Controller {
```
Comment (evert) [inline src/group/controller/collection.ts:23]: The most important endpoint is probably the `put` on this resource, letting us update the list of members. This is more consistent with everything else. The form is also nice though, as it lets us easily make changes.

**Comment by juhangsin** [comment]
As suggested, I'm gonna focus on the API call for now, and I'll add the proper form after.

File: src/group/service.ts [hunk 2]
```diff
@@ -30,3 +32,22 @@ export async function findAllGroupMembers(user: User): Promise<User[]> {
   return models;
 
 }
+
+export async function findByUserId(userId: number): Promise<any> {
```
Comment (evert) [inline src/group/service.ts:36]: This method duplicates `findAllGroupMembers` . Why make another?

File: src/group/service.ts [hunk 3]
```diff
@@ -30,3 +32,22 @@ export async function findAllGroupMembers(user: User): Promise<User[]> {
...
+  const query = 'SELECT * FROM group_members WHERE user_id = ?';
+  const result = await database.query(query, [userId]);
+
+  if (result[0].length !== 1) {
+    throw new NotFound(`Can't find group member with User ID ${userId}`);
+  }
+
+}
+
+export async function save (userId: string, group: User): Promise<void> {
```
Comment (evert) [inline src/group/service.ts:47]: I would expect a `save` function in a group save a group. This does something very different.

File: src/group/service.ts [hunk 4]
```diff
@@ -30,3 +32,22 @@ export async function findAllGroupMembers(user: User): Promise<User[]> {
...
+
+  const query = `INSERT INTO group_members SET group_id = ${group.id}, user_id = ?`;
```
Comment (evert) [inline src/group/service.ts:49]: SQL injection

File: src/group/controller/collection.ts (hunk 1 above)
Comment (evert) [reply src/group/controller/collection.ts:24]: This has not been addressed yet. We usually update resources with `PUT` and the request body should match what we got with `GET`

File: src/group/controller/collection.ts [hunk 5]
```diff
@@ -14,13 +15,39 @@ class GroupMemberCollectionControlller extends Controller {
...
+      if (!(err instanceof NotFound)) {
+        throw err;
+      }
+    }
+
+    await groupService.save(userBody.userId, group);
+
+    ctx.response.status = 201;
+    ctx.response.headers.set('Location', '/user/' + group.id + '/member');
+  }
```
Comment (evert) [inline src/group/controller/collection.ts:48]: This suggests that a new resource has been created with this location, and that's not really true.

File: src/group/type.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export type GroupMember = {
```
Comment (juhangsin) [inline src/group/type.ts:1]: i guess this type isn't being used at all, so we could remove it

Comment (evert) [reply src/group/type.ts]: Removed!

**Review by juhangsin — COMMENTED** [review]
:+1:

**Pull request merged**